junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertionsTests.java

```diff
@@ -2743,6 +2746,406 @@ void assertArrayEqualsDifferentNestedObjectArraysAndMessageSupplier() {
 		}
 	}
 
+	// --- assertIterableEquals --------------------------------------------
+
+	private static final class I implements Iterable<Object> {
+
+		private final List<Object> elements;
+
+		static I of(Object... elements) {
+			return new I(elements);
+		}
+
+		private I(Object... elements) {
+			this.elements = asList(elements);
+		}
+
+		@Override
+		public Iterator<Object> iterator() {
+			return elements.iterator();
+		}
+
+	}
+
+	@Test
+	void assertIterableEqualsObjectIterablesAndObjectLists() {
```

Comment:
Thank you very much for your suggestions here.

I've gone and split up this test method into smaller, more logical ones on my local machine, and I've also changed `I` into `IterableFactory` and implemented `listOf(Object...)` and `setOf(Object...)`. However I've decided against `arrayOf(Object...)`, since to the best of my knowledge arrays don't implement `Iterable`.

I aim to push a set of commits containing all the changes you suggested to this fork soon.